src/Inventory/Asset/VirtualMachine.php

```diff
@@ -302,36 +301,53 @@ protected function createVmComputer()
                      $computers_vm_id = $data['id'];
                 }
                 if ($computers_vm_id == 0) {
-                   // Add computer
-                    $vm->entities_id = $this->item->fields['entities_id'];
-                    $computers_vm_id = $computervm->add(Toolbox::addslashes_deep((array)$vm));
+                    //call rules on current collected data to find item
+                    //a callback on rulepassed() will be done if one is found.
+                    $rule = new RuleImportAssetCollection();
+                    $rule->getCollectionPart();
+                    $input = (array)$vm;
+                    $input ['itemtype'] = \Computer::class;
+                    $input['entities_id'] = $this->main_asset->getEntityID();
+                    $input  = \Toolbox::addslashes_deep($input);
+                    $datarules = $rule->processAllRules($input);
+
+                    if (isset($datarules['_no_rule_matches']) and ($datarules['_no_rule_matches'] == '1') or isset($datarules['found_inventories'])) {
+                        //this is a new one
+                        $vm->entities_id = $this->item->fields['entities_id'];
+                        $computers_vm_id = $computervm->add($input);
+                    } else {
+                        //refused by rules
+                        return;
+                    }
+                    // Add computer
```

Comment:
Should this comment be placed somewhere else ?

Reply:
(I've removed this one)